### PR TITLE
DiskRP swagger change for api version 2020-06-30

### PR DESCRIPTION
**Pull request by laurawu19**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json [hunk 1]
```diff
@@ -1634,6 +1634,10 @@
         "diskAccessId": {
           "type": "string",
           "description": "ARM id of the DiskAccess resource for using private endpoints on disks."
+        },
+        "tier": {
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json:1599]: tier should be supported in update too? also is tier supported in snapshots? Please confirm with Rahul

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json [hunk 2]
```diff
@@ -2089,6 +2119,11 @@
           "type": "integer",
           "format": "int64",
           "description": "If createOption is Upload, this is the size of the contents of the upload including the VHD footer. This value should be between 20972032 (20 MiB + 512 bytes for the VHD footer) and 35183298347520 bytes (32 TiB + 512 bytes for the VHD footer)."
+        },
+        "logicalSectorSize": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Support customers creating ultra disks with logicalSectorSize 512E. Support values are 512 and 4096 (default value)."
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json:2134]: typo: Supported values are

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json [hunk 3]
```diff
@@ -1862,6 +1866,28 @@
...
+        "EncryptionAtRestWithCustomerKey",
+        "EncryptionAtRestWithPlatformAndCustomerKeys"
+      ],
+      "x-ms-enum": {
+        "name": "DiskEncryptionSetType",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "EncryptionAtRestWithCustomerKey",
+            "description": "Disk is encrypted at rest with Customer managed key that can be changed and revoked by a customer."
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json:1847]: descriptions need to be updated to something like Resources using the disk encryption set would be encrypt at rest ...

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json [hunk 4]
```diff
@@ -2089,6 +2119,11 @@
           "type": "integer",
           "format": "int64",
           "description": "If createOption is Upload, this is the size of the contents of the upload including the VHD footer. This value should be between 20972032 (20 MiB + 512 bytes for the VHD footer) and 35183298347520 bytes (32 TiB + 512 bytes for the VHD footer)."
+        },
+        "logicalSectorSize": {
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json:2131]: please add example where logicalSectorSize is present in Get and also example where customer should be able to update an existing disk with different logicalSectorSize

Comment (laurawu19) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json:2131]: Added example which create an ultra disk with logicalSectorSize. Customer cannot update logicalSectorSize with different value.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json (hunk 1 above)
Comment (laurawu19) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json:1599]: Already added. And sync with Rahul, tier is not supported in snapshots.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json (hunk 3 above)
Comment (laurawu19) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json:1847]: Updated.

**Comment by qwordy** [comment]
You need to implement something like this https://docs.microsoft.com/en-us/rest/api/compute/operations/list

**Comment by laurawu19** [comment]
> Maybe you need to implement something like this https://docs.microsoft.com/en-us/rest/api/compute/operations/list according to the CI errors.

Have synced with swagger team, this error could be ignored.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json [hunk 5]
```diff
@@ -1634,6 +1634,10 @@
         "diskAccessId": {
           "type": "string",
           "description": "ARM id of the DiskAccess resource for using private endpoints on disks."
+        },
+        "tier": {
+          "type": "string",
+          "description": "Customer visible tier on the disk which indicates the tier to be set on the blob."
```
Comment (vikramd-ms) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json:1649]: Performance tier of the disk (e.g, P4, S10) as described here: https://azure.microsoft.com/en-us/pricing/details/managed-disks/. Does not apply to Ultra disks

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json [hunk 6]
```diff
@@ -1862,6 +1866,28 @@
...
+        "name": "DiskEncryptionSetType",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "EncryptionAtRestWithCustomerKey",
+            "description": "Disk is encrypted at rest with Customer managed key that can be changed and revoked by a customer."
+          },
+          {
+            "value": "EncryptionAtRestWithPlatformAndCustomerKeys",
+            "description": "Disk is encrypted at rest with 2 layers of encryption. One of the keys is Customer managed and the other key is Platform managed."
```
Comment (vikramd-ms) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json]: 2 -> two

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json (hunk 2 above)
Comment (vikramd-ms) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json]: Logical sector size for Ultra disks. Supported values are 512 and 4096. Default value is 4096.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json [hunk 7]
```diff
@@ -0,0 +1,2754 @@
...
+        },
+        "uploadSizeBytes": {
+          "type": "integer",
+          "format": "int64",
+          "description": "If createOption is Upload, this is the size of the contents of the upload including the VHD footer. This value should be between 20972032 (20 MiB + 512 bytes for the VHD footer) and 35183298347520 bytes (32 TiB + 512 bytes for the VHD footer)."
+        },
+        "logicalSectorSize": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Support customers creating ultra disks with logicalSectorSize 512E. Supported values are 512 and 4096 (default value)."
```
Comment (vikramd-ms) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json:2182]: Logical sector size in bytes for Ultra disks. Supported values are 512 ad 4096. 4096 is the default.

**Review by ravbhatnagar — COMMENTED** [review]
Signing off from ARM side.

**Review by haagha — APPROVED** [review]
Looks good from Cplat Compute side. @laurawu19 please ask someone from OneSDK team to suppress CI errors.

**Comment by haagha** [comment]
@yungezz Is the DoNotMerge label ready to be removed? 
And if the CI check failure Isn't significant can you suppress it so this can be merged.

**Pull request merged**